### PR TITLE
Add ztl to main build workflow

### LINKED ISSUE #43: ZTL workflow maintenance
One bug and some little needed maintenance tasks for zoning tax lots github action workflow files.
- [x] [Dataloading action is failing](https://github.com/NYCPlanning/data-engineering/actions/workflows/zoningtaxlots_dataloading.yml). This is a step separate from the build, one that uses the [published library archive action](https://github.com/NYCPlanning/action-library-archive) to archive various needed inputs for ztl build. GIS generally runs this action since they are responsible for updating the source datasets. It's failing because the filepath for template files was not updated when I initially went over ztl in the monorepo to get it building.
- [x] build action is not using build docker image. Workflow file should be updated to use nycplanning/build-base image (see [pluto](https://github.com/NYCPlanning/data-engineering/blob/main/.github/workflows/pluto_build.yml#L49) for example, also for the env setup step)
- [x] build and dataloading actions both should be updated to ubuntu 22.04, checkout action v3
- [x] build action should be updated to postgis 14-3.3
- [x] `close issue` job of dataloading can be deleted, as well as other logic around triggering workflow via issue
- add ztl as an option in the generic `build` workflow file. this requires both
  - [x] changes to the "generic" build action as well as
  - [x] addition of `workflow call` as option for ztl build workflow file

And as part of the above
- [ ] qc outputs of a build after changes have been made

### PR DESCRIPTION
**Pull request by fvankrieken**
Closes #43

### CODE REVIEW

File: .github/workflows/zoningtaxlots_build.yml
Comment (damonmcc) [inline .github/workflows/zoningtaxlots_build.yml:1]: part of the linked issue is to have this use the build docker image, I think by adding `container: image: nycplanning/build-base:latest` and dropping the `install dependencies` step

**Comment by damonmcc** [comment]
just made #163 to fix that template unit test and make it more stable

File: .github/workflows/zoningtaxlots_build.yml
Comment (fvankrieken) [reply .github/workflows/zoningtaxlots_build.yml:1]: Oh oops - thank you

**Comment by fvankrieken** [comment]
Successful build (minus upload)
https://github.com/NYCPlanning/data-engineering/actions/runs/5891162968

File: .github/workflows/zoningtaxlots_build.yml [hunk 1]
```diff
@@ -23,30 +29,26 @@ jobs:
...
+        run: ./bash/docker_container_setup.sh
+        
       - name: 1. dataloading ...
         run: ./ztl.sh dataloading
 
       - name: 2. build ...
         run: ./ztl.sh build
 
-      - name: 3. qaqc ...
+      - name: 3. qaqc, export, and upload ...
```
Comment (fvankrieken) [inline .github/workflows/zoningtaxlots_build.yml:52]: Would have split these into multiple bash scripts but for ztl they're kind of intertwined

**Pull request merged**